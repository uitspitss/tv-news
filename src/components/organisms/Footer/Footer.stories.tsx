import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import base from 'paths.macro';

import { Footer, FooterProps } from './Footer';

const path = base.replace(
  /^.+\/(atoms|molecules|organisms|templates|pages)\/(.+)\//,
  '$1/$2',
);

export default {
  title: `${path}`,
  component: Footer,
} as Meta;

const Template: Story<FooterProps> = (args) => <Footer {...args} />;

export const Desktop = Template.bind({});

export const Mobile = Template.bind({});
Mobile.parameters = {
  viewport: {
    defaultViewport: 'iphonex',
  },
};
