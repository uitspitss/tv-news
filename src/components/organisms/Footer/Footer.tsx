import React, { memo, VFC } from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import tw, { styled } from 'twin.macro';

import SEO from '~/next-seo.config.json';
import { copyrightYears } from '~/utils/copyright-years';

export type FooterProps = {
  /**
   * class for twin.macro
   */
  className?: string;
};

type Props = FooterProps;

const Component: VFC<Props> = (props) => {
  const { className } = props;

  return (
    <div className={className}>
      <p className="title">
        ©{copyrightYears(2021)} — {SEO.title}
      </p>
      <a
        className="twitter"
        href="https://twitter.com/uitspitss"
        rel="noopener noreferrer"
        target="_blank"
      >
        @uitspitss
      </a>
    </div>
  );
};

const StyledComponent = styled(Component)`
  & {
    ${tw`flex items-center py-2 px-2 lg:px-6 bg-primary-700`}

    & > .title {
      ${tw`text-sm text-primary-500 mr-1`}
    }

    & > .twitter {
      ${tw`text-sm text-primary-500 no-underline`}
    }
  }
`;

/**
 * footer
 */
export const Footer = memo((props: FooterProps) => (
  <StyledComponent {...props} />
));
