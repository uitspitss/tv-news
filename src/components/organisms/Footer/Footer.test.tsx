import React from 'react';
import { render } from '@testing-library/react';

import { Footer } from './Footer';

describe('Footer', () => {
  test('snapshot test', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
