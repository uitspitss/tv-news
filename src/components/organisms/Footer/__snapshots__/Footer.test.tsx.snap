// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot test 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  --tw-bg-opacity: 1;
  background-color: rgba(55, 65, 81, var(--tw-bg-opacity));
}

@media (min-width: 1024px) {
  .emotion-0 {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
  }
}

.emotion-0>.title {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(107, 114, 128, var(--tw-text-opacity));
  margin-right: 0.25rem;
}

.emotion-0>.twitter {
  font-size: 0.875rem;
  line-height: 1.25rem;
  --tw-text-opacity: 1;
  color: rgba(107, 114, 128, var(--tw-text-opacity));
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <p
      class="title"
    >
      ©
      2021
       — 
      TV局ニュースプレイリスト
    </p>
    <a
      class="twitter"
      href="https://twitter.com/uitspitss"
      rel="noopener noreferrer"
      target="_blank"
    >
      @uitspitss
    </a>
  </div>
</div>
`;
